1) With conservation_consensus_code.py, conservation scores and consensus sequences of the different data can be obtained.
2) With txtTOtsvfileforFirst.py, txtTOtsvfileforSecond.py, and txtTOtsvfileforThird.py; tsv files of consensus amino acid's conservation score per position can be obtained. 
3) With subtreetofastasequences.py, nwk file of the subtree can be turned into a fasta file which contains sequences information to then realign and form a tree.
4) With bootstrapbiggerthan70.py, subtree can be pruned according to Bootstrap values and can obtain the fasta file of the filtered sequences.
5) With map_mutations_positions.py, original sequence of chosen Homo Sapiens isoform can be mapped to aligned files.
6) With known_mutations.py, threshold values for classification can be calculated.
7) With gnomadVarAnalysis.py, classification of unknown variants can be performed.
8) With ttest.py, t-test to get p-values to determine significance can be performed.
9) With cons_per_position.py, conservation scores per position scatter plot with a line plot overlaid can be formed.
10) With output.R, conservation scores and allele frequency scatter plot of mutations can be formed.
11) With conservationhistogram.py, conservation score information to histogram table can be formed.
   
